marklogic-data-hub/src/test/java/com/marklogic/hub/impl/ClearUserModulesTest.java

```diff
@@ -0,0 +1,126 @@
+package com.marklogic.hub.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.admin.QueryOptionsManager;
+import com.marklogic.client.admin.ResourceExtensionsManager;
+import com.marklogic.client.admin.ServerConfigurationManager;
+import com.marklogic.client.admin.TransformExtensionsManager;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.QueryOptionsListHandle;
+import com.marklogic.hub.AbstractHubCoreTest;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
+
+class ClearUserModulesTest extends AbstractHubCoreTest {
+
+    @Test
+    void testClearUserModules() {
+        new DataHubImpl(runAsDataHubDeveloper()).clearUserModules();
+        Assumptions.assumeTrue(isVersionCompatibleWith520Roles());
+        ModuleCounts beforeClearingModules = new ModuleCounts();
+        beforeClearingModules.setModuleCounts();
+
+        Assertions.assertEquals(1, beforeClearingModules.optionsCount);
+        Assertions.assertEquals(12, beforeClearingModules.transformsCount);
+        Assertions.assertEquals(23, beforeClearingModules.resourceExtensionCount);
+
+        installProjectWithEachTypeOfModule();
+
+        try {
+            new DataHubImpl(runAsDataHubOperator()).clearUserModules();
+            Assertions.fail("'data-hub-operator' should not be able delete user modules");
+        } catch (RuntimeException e) {
+            Assertions.assertTrue(e.getMessage().contains("User is not allowed to delete /config/query"));
+            logger.info("'data-hub-operator' cannot delete user modules");
+        }
+
+        ModuleCounts afterClearingModulesAsOperator = new ModuleCounts();
+        afterClearingModulesAsOperator.setModuleCounts();
+
+        Assertions.assertEquals(2, afterClearingModulesAsOperator.optionsCount);
```

Comment:
These assertions should then be relative to the before counts - e.g. optionsCount = before + 1.